test/functional/rpc_decodescript.py

```diff
@@ -252,13 +255,23 @@ def decoderawtransaction_asm_sighashtype(self):
         rpc_result = self.nodes[0].decoderawtransaction(txSave.serialize().hex())
         assert_equal('OP_RETURN 3011020701010101010101020601010101010101', rpc_result['vin'][0]['scriptSig']['asm'])
 
+    def decodescript_datadriven_tests(self):
+        with open(os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data/rpc_decodescript.json'), encoding='utf-8') as f:
+            dd_tests = json.load(f)
+
+        for dd_test in dd_tests:
```

Comment:
nit: If you make `dd_tests` a dictionary of the form `{ 'input1': { ... rpc result ... }, 'input2': {...}, ...}`, you can avoid the array indices.

```suggestion
        for script, result in dd_tests:
```

Reply:
I think that should work if you don't change the data type? `,` would unpack the arrays.
When converting to a dictionary you'd need to iterate over `dd_tests.items()`.